src/tls/server.c

```diff
@@ -552,45 +553,53 @@ server_init (const char *ws_path,
     }
 
   /* systemd socket activated? */
-  listen_fds = secure_getenv ("LISTEN_FDS");
-  if (listen_fds && check_sd_listen_pid ())
+  env_listen_fds = secure_getenv ("LISTEN_FDS");
+  if (env_listen_fds && check_sd_listen_pid ())
     {
-      if (strcmp (listen_fds, "1") != 0)
-        errx (1, "This program can only accept exactly one socket from systemd, but got passed %s", listen_fds);
-      server.listen_fd = SD_LISTEN_FDS_START;
-      debug ("Server ready. Listening to systemd activated socket fd %i", server.listen_fd);
+      char *endptr = NULL;
+      long n = strtol (env_listen_fds, &endptr, 10);
+      if (n < 1 || n > MAX_LISTEN_FDS || !endptr || *endptr != '\0')
+        errx (1, "Invalid $LISTEN_FDS value '%s'; this program supports up to 10 fds", env_listen_fds);
+      for (int i = 0, fd = SD_LISTEN_FDS_START; i < n; ++i, ++fd)
+        {
+          server.listen_fds[i] = fd;
+          debug ("Listening to systemd activated socket fd %i", fd);
+        }
```

Comment:
please mark EOF (if <10) with -1 rather than 0.  0 is a valid fd.

Reply:
Seems odd to not have stdin, but indeed that's better. fixed.